src/bin/electrs.rs

```diff
@@ -88,7 +85,7 @@ fn run_server(config: Arc<Config>) -> Result<()> {
         &metrics,
         Arc::clone(&config),
     )));
-    mempool.write().unwrap().update(&daemon)?;
+    Mempool::update(&mempool, &daemon)?;
```

Comment:
Not directly related to this PR, but I wonder if we should perhaps make the initial mempool sync run in a loop until it succeeds?

Reply:
That might be messy. I don't think this will work:

```
    loop {
        match Mempool::update(&mempool, &daemon) {
            Ok(_) => break,
            Err(e) => {
                println!("Error updating mempool, trying again in 5 seconds: {}", e);
                thread::sleep(Duration::from_secs(5));
            },
        }
    }
```

because the loop also has to explicitly await a server kill signal (like this):
```
if let Err(err) = signal.wait(Duration::from_secs(5), true) {
    ...
}
```

What if we don't fail when the update fail. Instead we could print out an error message,  leave the mempool empty. and just continue to the main loop (which will try to update the mempool every 5 seconds)?

Reply:
The mempool loop could also use `signal.wait()`, which awaits both the timeout and kill signal (whichever happens first) and returns an `Ok` if the timeout was hit or an `Err` if it was the kill signal. Something like this should work:

```rust
loop {
    match Mempool::update(&mempool, &daemon) {
        Ok(_) => break,
        Err(e) => {
            println!("Error updating mempool, trying again in 5 seconds: {}", e);
            signal.wait(Duration::from_secs(5))?; // could be a shorter duration here
        },
    }
}
```

> What if we don't fail when the update fail. Instead we could print out an error message, leave the mempool empty. and just continue to the main loop (which will try to update the mempool every 5 seconds)?

This is the current behavior, `update()` returns an `Ok()` when it fails downloading the mempool transactions so it'll just move on. I was thinking that it might be worth making sure that we have an available mempool before starting up.